internal/git/git_test.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2021 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package git
+
+import (
+	"testing"
+
+	"github.com/go-git/go-git/v5/utils/merkletrie"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/thriftrw/internal/breaktest"
+)
+
+func TestOpenRepo(t *testing.T) {
+	t.Parallel()
+	tmpDir := t.TempDir()
+	breaktest.CreateRepoAndCommit(t, tmpDir)
```

Comment:
Looking at how CreateRepoAndCommit is used in this test, I think that CreateRepoAndCommit hides too much of the details of what files get changed. In this package's test, we need to look into the internals of CreateReportAndCommit, which are in a different package,  to know what values to assert.

It seems like CreateRepoAndCommit should take as inputs file modifications to commit. With this, all the knowledge of what get committed and asserted can live in the same test and package.